bitnami/janusgraph/templates/_helpers.tpl

```diff
@@ -0,0 +1,277 @@
+{{- end -}}
+{{- end -}}
+
+{{/*
+Returns the hostname of the configured storage backend
+*/}}
+{{- define "janusgraph.storage.hostname" -}}
+{{- if .Values.storageBackend.cassandra.enabled -}}
+{{- include "common.names.dependency.fullname" (dict "chartName" "cassandra" "chartValues" .Values.storageBackend.cassandra "context" $) -}}
+{{- else if .Values.storageBackend.external.hostname -}}
+{{- print .Values.storageBackend.external.hostname -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Returns the port of the configured storage backend
+*/}}
+{{- define "janusgraph.storage.port" -}}
+{{- if .Values.storageBackend.cassandra.enabled }}
+{{- printf "%d" (int .Values.cassandra.service.ports.cql) -}}
+{{- else if .Values.storageBackend.external.port -}}
+{{- printf "%d" (int .Values.storageBackend.external.port) -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Create the storage password secret name
+*/}}
+{{- define "janusgraph.storage.username" -}}
+{{- if .Values.storageBackend.cassandra.enabled -}}
+{{- print (default "cassandra" .Values.cassandra.dbUser.user) -}}
+{{- else if .Values.storageBackend.external.username -}}
+{{- .Values.storageBackend.external.username -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Create the storage password secret name
+*/}}
+{{- define "janusgraph.storage.password.secretName" -}}
+{{- if .Values.storageBackend.cassandra.enabled -}}
+{{- printf "%s-cassandra" (include "common.names.fullname" .) -}}
+{{- else if .Values.storageBackend.external.existingSecret -}}
+{{- .Values.storageBackend.external.existingSecret -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Create the storage password secret key
+*/}}
+{{- define "janusgraph.storage.password.secretKey" -}}
+{{- if .Values.storageBackend.cassandra.enabled -}}
+cassandra-password
+{{- else if .Values.storageBackend.external.existingSecretPasswordKey -}}
+{{- .Values.storageBackend.external.existingSecretPasswordKey -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Returns true if a index management backend has been configured
+*/}}
+{{- define "janusgraph.index.enabled" -}}
+{{- if or .Values.indexBackend.lucene.enabled .Values.indexBackend.external.backend -}}
+{{- true -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Returns the configured index management backend
+*/}}
+{{- define "janusgraph.index.backend" -}}
+{{- if .Values.indexBackend.lucene.enabled -}}
+{{- print "lucene" -}}
+{{- else if .Values.indexBackend.external.backend -}}
+{{- print .Values.indexBackend.external.backend -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Returns the hostname of the configured index management backend
+*/}}
+{{- define "janusgraph.index.hostname" -}}
+{{- if .Values.indexBackend.external.hostname -}}
+{{- print .Values.indexBackend.external.hostname -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Returns the port of the configured index management backend
+*/}}
+{{- define "janusgraph.index.port" -}}
+{{- if .Values.indexBackend.external.port -}}
+{{- printf "%d" ( int .Values.indexBackend.external.port) -}}
+{{- end -}}
+{{- end -}}
```

Comment:
No "else" conditions on these cases?

Reply:
No 'else' condition at the moment because only two indexing methods are supported:
- Lucene (local)
- External

The else would be added in the future, in case we add support for other indexing methods using subcharts, such as elasticsearch or Apache Solr.

Reply:
Then there's no need for the `if`, right?

Reply:
Well, both `.Values.indexBackend.external.port` and `.Values.indexBackend.external.hostname` could be empty

Reply:
In this case I don't see the benefit of using the `default` function

I would rather keep it as it is, so it maintains the similarities with storage backend logic, and in case we decide to add indexing subcharts in the future we would only need to add if-else statements

Reply:
It's cleaner but I'll leave it up to you